src/pipeline/pipelineExplorer.ts

```diff
@@ -19,7 +20,7 @@ export class PipelineExplorer implements TreeDataProvider<TektonNode>, Disposabl
 
   constructor() {
     this.fsw = WatchUtil.watchFileForContextChange(kubeConfigFolder, 'config');
-    this.fsw.emitter.on('file-changed', this.refresh.bind(this));
+    this.fsw.emitter.on('file-changed', this.contextChange.bind(this));
```

Comment:
I think it would be better if you extract this file watcher in separate module, and add API to receive events when `config` file changed. Add listener for that event in this class and in `watchResources`. As it is not `PipelineExplorer ` responsibility to add file watcher and disable watching on tekton resources.